source/extensions/common/redis/cluster_refresh_manager_impl.cc

```diff
@@ -73,14 +73,25 @@ bool ClusterRefreshManagerImpl::onEvent(const std::string& cluster_name, EventTy
       if (threshold <= 0) {
         return false;
       }
+
+      bool postCallBack = false;
       // ignore redirects during min time between triggering
       if ((++(*count) >= threshold) &&
           (info->last_callback_time_ms_.compare_exchange_strong(last_callback_time_ms, now))) {
         // last_callback_time_ms_ successfully updated without any changes since it was
         // initially read. This thread is allowed to post a call to the registered callback
         // on the main thread. Otherwise, the thread would be ignored to prevent over-triggering
         // cluster callbacks.
+        postCallBack = true;
+      }
+
+      // If a callback should be triggered(in this or some other threads) signaled by the changed
+      // last callback time, we reset the count to 0
+      if (postCallBack || info->last_callback_time_ms_.load() != last_callback_time_ms) {
```

Comment:
I'm still a little confused here by this logic so I think it needs some more comments. What is the exact sequence that leads to post_callback being false but the times being not equal?

Reply:
ok, let me add those to the comments